### PR TITLE
Use real npmjs.org couchapp for self-test

### PR DESCRIPTION
**Pull request by rmg**
Rather than a comment saying how/where to get it, just does it.

Also fixes a dependency I found while testing with my empty `~/.npmrc`.

@bajtos this will conflict with at least one of your open PRs..

### CODE REVIEW

File: test/runner.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ var registryUrlPromise = couchRegistry.start();
 
 describe('The npmjs.org registry', function() {
   validate(registryUrlPromise, {
-    userCredentials: couchRegistry.userCredentials
+    userCredentials: couchRegistry.userCredentials,
+    userEmail: 'test@registry-validator'
```
Comment (bajtos) [inline test/runner.js]: The default userEmail should be rather set in the config defaults:
https://github.com/strongloop/registry-validator/blob/master/lib/config.js#L6

**Comment by bajtos** [comment]
The change looks good except the one comment above.

Don't worry about conflicts, I'll update my PRs after this change is landed.

**Comment by rmg** [comment]
`lib/config.js` would apply to all runs of the validator, not just the self test. I figured the self tests would double as an example of the required parameters. 

Another thought occurs to me.. The npmrc files on the system (global, user, etc.) should all be ignored/blocked by the internal npm instance so that external state affects the tests as little as possible.

**Comment by bajtos** [comment]
> `lib/config.js` would apply to all runs of the validator, not just the self test. I figured the self tests would double as an example of the required parameters. 

I'd rather keep the number of required parameters as low as possible. The default config should have sensible defaults in as many places as possible.

> Another thought occurs to me.. The npmrc files on the system (global, user, etc.) should all be ignored/blocked by the internal npm instance so that external state affects the tests as little as possible.

Yes, that would be great. In Reggie [tests](https://github.com/mbrevoort/node-reggie/blob/80ba9ade4a69c432b931e6be2384d16a7a82f5de/test/helpers.js#L70-L108), we are creating a temporary `.npmrc` file. Which is a bit of extra work.

Perhaps if npm was handling a missing npmrc file gracefully, then we could add a config option pointing to a non-existent config file in order to disable user config file?

**Comment by rmg** [comment]
I've moved the email to `lib/config.js` and I've disabled loading of the host environment's `~/.npmrc` and `$PREFIX/etc/npmrc`. I'll squash/cleanup before merging.

**Comment by bajtos** [comment]
Hopefully the config filenames are unique enough to not exist on the user file system.

LGTM, please squash and merge.
